app/assets/javascripts/govuk_publishing_components/components/add-another.js

```diff
@@ -0,0 +1,159 @@
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {};
+
+(function (Modules) {
+  function AddAnother (module) {
+    this.module = module
+  }
+
+  AddAnother.prototype.init = function () {
+    this.initialiseClickEvent()
+    this.preventFormSubmissions()
+  }
+
+  AddAnother.prototype.initialiseClickEvent = function () {
+    this.module.addEventListener(
+      'click',
+      function (e) {
+        if (e.target.classList.contains('js-add-another__remove-button')) {
+          this.removeFields(e.target)
+        } else if (e.target.classList.contains('js-add-another__add-button')) {
+          this.addFields(e.target)
+        }
+      }.bind(this)
+    )
+  }
+
+  AddAnother.prototype.preventFormSubmissions = function () {
+    var addButton = this.module.querySelector('.js-add-another__add-button')
+    var removeButtons = this.module.querySelectorAll('.js-add-another__remove-button')
+    addButton.setAttribute('type', 'button')
+    removeButtons.forEach(function (removeButton) {
+      removeButton.setAttribute('type', 'button')
+    })
+  }
+
+  AddAnother.prototype.addFields = function (button) {
+    var allFields = button.parentNode.querySelectorAll(
+      '.js-add-another__repeated-fields'
+    )
+    var fields = allFields[allFields.length - 1]
+
+    // Show hidden "Remove" button
+    if (fields.querySelector('.js-add-another__remove-button')) {
+      fields.querySelector('.js-add-another__remove-button').style.display =
+                'inline-block'
+    }
+
+    // Clone the markup of the previous set of fields
+    var newFields = fields.cloneNode(true)
+
+    // Reset values of cloned fields
+    newFields
+      .querySelectorAll('input, textarea, select')
+      .forEach(function (element) {
+        element.value = ''
+      })
+
+    // Increment values for id, for and name of cloned fields
+    this.setValues(newFields, null)
+
+    // Add cloned fields to the DOM
+    button.before(newFields)
```

Comment:
It strikes me that there should probably be an announcement for screen reader users that content has been added to the DOM. Would appreciate some guidance on this from somebody more knowledgeable in the area.